### ISSUE TITLE
Remove unnecessary generics in query builder functions

### ISSUE THREAD

**Issue by olee** [supabase/postgrest-js#519]
As far as I can see, functions like `insert` and `upsert` in the query builder use a generic type `Row` for the values to insert when in fact this type should not be necessary at all.

So instead of this:
https://github.com/supabase/postgrest-js/blob/c9cebf8a0a5a3c8dff4a511d70fd53079781f563/src/PostgrestQueryBuilder.ts#L101-L106

It should work better if it is defined like this instead:
```ts
 insert( 
   values: Relation extends { Insert: unknown } ? Relation['Insert'] : never,
   options?: { 
     count?: 'exact' | 'planned' | 'estimated' 
   } 
 ): PostgrestFilterBuilder<Schema, Relation['Row'], null, RelationName, Relationships> 
```

This would improve type discovery a bit and improve error reporting.

The reason I say it's not required is because the generic type is only used for the input and not used anywhere on the output.

**Comment by jrmoynihan** [comment]
Agreed. Overloads often don't play well with Typescript language server, in my experience.
